src/mongoose_account_api.erl

```diff
@@ -0,0 +1,327 @@
+%% @doc Provide an interface for frontends (like graphql or ctl) to manage accounts.
+-module(mongoose_account_api).
+
+-export([list_users/1,
+         register_user/3,
+         register_generated_user/2,
+         unregister_user/1,
+         unregister_user/2,
+         delete_old_users/1,
+         delete_old_users_for_domain/2,
+         ban_account/2,
+         ban_account/3,
+         change_password/2,
+         change_password/3,
+         check_account/1,
+         check_account/2,
+         check_password/2,
+         check_password/3,
+         check_password_hash/3,
+         check_password_hash/4,
+         num_active_users/2]).
+
+-include("mongoose.hrl").
+
+-type register_result() :: {ok | exists | invalid_jid | cannot_register, string()}.
+
+-type unregister_result() :: {ok | not_allowed | invalid_jid, string()}.
+
+-type change_password_result() :: {ok | empty_password | not_allowed | invalid_jid, string()}.
+
+-type check_password_result() :: {ok | incorrect | user_does_not_exist, string()}.
+
+-type check_password_hash_result() :: {ok | incorrect | error, string()}.
+
+-type check_account_result() :: {ok | user_does_not_exist, string()}.
+
+-type num_active_users_result() :: {ok, non_neg_integer()} | {cannot_count, string()}.
+
+-type delete_old_users() :: {ok, string()}.
```

Comment:
I think this will be more consistent:

```suggestion
-type delete_old_users_result() :: {ok, string()}.
```
The references to it would need fixing as well